### ISSUE TITLE
PDF-UA AccessibilityProperties for building a tag-tree

### ISSUE THREAD

**Issue by tabibite** [LibrePDF/OpenPDF#181]
I would like to create a tagged PDF and there I need to build a tag-tree. Is it possible with OpenPDF?

**Comment by hvbtup** [comment]
Eclipse BIRT uses OpenPDF for PDF generation. This project would also benefit from PDF/UA support in OpenPDF.

I think this is one important open source topic where the big players, governments, and the EU should pay some developer(s) for developing this.

**Comment by christophersavory** [comment]
@asturio, @bengolder, @daviddurand, @kulatamicuda, @rtfarte, @tlxtellef how difficult is it to add the tag-tree to OpenPDF?

**Comment by hvbtup** [comment]
AFAIK JasperReports already supports creating PDF/UA using OpenPDF.
_Can somebody confirm this?_
So it seems like OpenPDF *can* already create PDF/UA if used in the right way, and it's just a matter of missing examples.

**Comment by christophersavory** [comment]
@asturio Can OpenPDF already create a PDF/UA document or is it a feature that needs to be built in?

**Comment by asturio** [comment]
It should be possible to create "navigable" PDFs with OpenPDF. But there is no example code for that. I think flyingsaucer (using OpenPDF) can do this too.

**Comment by anmeeks-vt-edu** [comment]
I want to chime in with the others that some example code would be very helpful.  I found a comment on S.O. where someone managed to add headings on a chunk using: 
```
Chunk heading = new Chunk("A guide to key terms\n " , font);
heading.setRole(PdfName.H1);
```
As a novice user of OpenPDF, I really need a kitchen sink example to pull from.  Reading order, headings, links, semantic lists, screen-reader-only text are a few things I know I'm already worried about.  PDFs created by the existing tutorials are not tagged at all.  If I have a pdf open in Acrobat and turn on NVDA, I get a warning message from Acrobat that the generated document is not tagged. NVDA also fails to read any text in the document even though it is there visually.  My organization is WCAG 2.1/AA compliant.  I can't use this in production till I can guarantee what it generates is accessible.
